sleap_roots/angle.py

```diff
@@ -0,0 +1,52 @@
+"""Get angle of each root."""
+
+import numpy as np
+import math
+
+
+def get_node_ind(pts: np.ndarray, proximal=True) -> np.ndarray:
+    """Find nproximal/distal node index.
+
+    Args:
+        pts: Numpy array of points of shape (instances, nodes, 2).
+        proximal: Boolean value, where true is proximal (default), false is distal.
+
+    Returns:
+        An array of shape (instances,) of proximal or distal node index.
+    """
+    node_ind = []
+    for i in range(pts.shape[0]):
+        ind = 1 if proximal else pts.shape[1] - 1  # set initial proximal/distal node
+        while np.isnan(pts[i, ind]).any():
+            ind += 1 if proximal else -1
```

Comment:
Maybe we want to use a `for` loop with a `break` or alternatively, add another condition to the `while` so we prevent infinite loops in the case where we have all NaNs?

Reply:
Add an if function to break the while loop.